torch/_dynamo/variables/lists.py

```diff
@@ -541,12 +548,13 @@ def call_method(
             assert not kwargs
             tx.output.side_effects.mutation(self)
             self.items = [args[0]] + list(self.items)
+            slices = slice(None, maxlen)
             result = ConstantVariable.create(None)
         else:
             result = super().call_method(tx, name, args, kwargs)
 
-        if self.maxlen.as_python_constant() is not None:
-            self.items = list(self.items)[-self.maxlen.as_python_constant() :]
+        if slices is not None and maxlen is not None and len(self.items) > maxlen:
+            self.items = list(self.items)[slices]
```

Comment:
Minor nit, but this would be better served by itertools.islice I'm guessing, but I am not sure it would compile properly, would it?

Reply:
Oh, then why do you cast it to `list()` right before then?

Reply:
Updated.